filebeat/config/config.go

```diff
@@ -26,9 +33,9 @@ type Config struct {
 
 type FilebeatConfig struct {
 	Prospectors  []*common.Config `config:"prospectors"`
-	SpoolSize    uint64           `config:"spool_size"`
+	SpoolSize    uint64           `config:"spool_size" validate:"min=1"`
 	PublishAsync bool             `config:"publish_async"`
-	IdleTimeout  time.Duration    `config:"idle_timeout"`
+	IdleTimeout  time.Duration    `config:"idle_timeout" validate:"nonzero,min=0"`
```

Comment:
Not sure if that is something that we should encourage. That above allows to set it for example to 0.001s which is almost like disabling it. 

One thing I just realised related to duration and ucfg: it should be `min=1s` and not `min=1` as if the unit is not know, it could be "anything".